src/assets/DockerCompose/docker_compose.md

```diff
@@ -0,0 +1,210 @@
+          volumes:
+            - ./testdump.sql:/docker-entrypoint-initdb.d/testdump.sql
+          restart: always
+     ```
+
+      Here, defining services in docker compose we use the keyword `services`. We have defined our service `database` configured with `image:mysql` and the `environment` variables required for it.
+      We can see it has a volumes keyword used here. We can mount local data to the docker container using the `volumes` keyword. In this context we have mapped `testdump.sql` which is in our
+      local root directory which runs as an entrypoint when the container starts. It means when the `database` service is started in a container, the `testdump.sql` is the first thing which is run inside the container.
+      The entrypoint script will create 4 users in the database. Add some sql commands inside of `testdump.sql`:
+
+      ```sql
+        CREATE TABLE users(userid INT PRIMARY KEY AUTO_INCREMENT, fullname VARCHAR(60), email VARCHAR(60));
+        INSERT INTO users (fullname, email) VALUES ('Sagar Gurung', 'sagar@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Bob Johnson', 'bob@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Eva White', 'eva@example.com');
+        INSERT INTO users (fullname, email) VALUES ('Michael Brown', 'michael@example.com');
+        ALTER USER 'root' IDENTIFIED WITH mysql_native_password BY 'root';
+        flush privileges;
+     ```
+  2. backend service
+
+      ```yml
+        backend:
+          image: sagar4321/demo-server
+          environment:
+            PORT: 3001
+            DBHOST: database
+            DBPORT: 3306
+            DBUSER: root
+            DBNAME: demodb
+            DBPASSWORD: root
+          restart: always
+          ports:
+            - "3001:3001"
+          depends_on:
+            - database
+      ```
+
+      Similarly, we define and configure our backend as above with the necessary `image:sagar4321/demo-server` (this image is built and pushed to docker hub which I created), `environment` and `port 3001` (port where the server listens)
+      Here, we have another keyword `depends_on` which has value `database` service. It sets the order in which services must start and stop. For above case, order `database` service starts before the `backend` service start and vice-versa.
+
+  3. frontend service
+
+      ```yml
+          frontend:
+            image: sagar4321/demo-web
+            environment:
+              SERVER_URL: http://0.0.0.0:3001
+            ports:
+              - 8080:8080
+            depends_on:
+              - backend
+      ```
+
+      Use a similar style of definition and configuration for the `frontend` service with `image: sagar4321/demo-web` (which I built and pushed to docker hub), `environment` and `port 8080` (where the `frontend` service listen).
+      But here the `frontend` service depends on `backend` service since it needs to be started before the `frontend` service.
```

Comment:
```suggestion
      But here the `frontend` service depends on the `backend` service since it needs to be started before the `frontend` service.
```